File: Cpp/Two Pointers/Medium/ContainerWithMostWater.cpp
/*
Intuition


Approach


Complexity
- Time complexity:

- Space complexity:

*/
#include <vector>

int maxArea(std::vector<int>& height) {
    int left = 0, right = height.size()-1;
    int maxHeight = 0;

    while(left < right){
        int h = std::min(height[left], height[right]) * (right - left);
        maxHeight = std::max(h, maxHeight);

        if(height[left] < height[right]){
            left++;
        }
        else{
            right--;
        }
    }
    return maxHeight;
}

File: Python/Two Pointers/Medium/ContainerWithMostWater.py
# Intuition
'''
'''

# Approach
'''
'''

# Complexity

# - Time complexity:
'''
'''

# - Space complexity:
'''
'''
from typing import List

def maxArea(self, height: List[int]) -> int:
    left, right = 0, len(height)-1
    maxHeight = 0
    while(left < right):
        h = min(height[left], height[right]) * (right-left)
        maxHeight = max(h, maxHeight)

        if(height[left] < height[right]):
            left += 1
        else:
            right -= 1

    return maxHeight